crates/sui-json-rpc/src/indexer_api.rs

```diff
@@ -425,11 +468,20 @@ impl<R: ReadApiServer> IndexerApiServer for IndexerApi<R> {
                 })?
                 .value;
 
-            Ok(Page {
-                data: vec![domain.to_string()],
-                next_cursor: None,
-                has_next_page: false,
-            })
+            let domain_name = domain.to_string();
+
+            let resolved_address = self
+                .resolve_name_service_address(domain_name.clone())
+                .await?;
+
+            if resolved_address.is_none() || resolved_address != Some(address) {
+                return Ok(result);
+            }
+            // TODO(manos): Discuss why is this even a paginated response.
+            // This API is always going to return a single domain name.
+            result.data.push(domain_name);
```

Comment:
don't we need to do a similar validation here as we do in the normal domain lookup?

Reply:
That's what's actually added. We're resolving the name (doing the validation) & if it matches the requested address, returns the domain!
That validation wasn't here before!